### PR TITLE
add IQL

### PR DESCRIPTION
**Pull request by Jiukaishi**

### CODE REVIEW

File: benchmark/torch/iql/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
++ Each experiment was run three times with different seeds
+
+## How to use
+### Dependencies:
++ python3.5+
++ [parl>2.0.4](https://github.com/PaddlePaddle/PARL)
++ torch
++ gym==0.20.0
++ mujoco-py==2.0.2.13
++ [d4rl](https://github.com/rail-berkeley/d4rl) (install from source)
```
Comment (zenghsh3) [inline benchmark/torch/iql/README.md]: Pls clarify the version of d4rl.

File: parl/algorithms/torch/iql.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        Args:
+            decay(float): the decaying factor while updating the target network with the training network.
+                        0 represents the **assignment**. None represents updating the target network slowly that depends on the hyperparameter `tau`.
+        """
+
+        for param, target_param in zip(self.model.parameters(),
+                                       self.q_target.parameters()):
+            target_param.data.copy_(alpha * param.data +
+                                    (1 - alpha) * target_param.data)
+        return None
```
Comment (zenghsh3) [inline parl/algorithms/torch/iql.py]: We can remove this line.

File: parl/algorithms/torch/iql.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+EXP_ADV_MAX = 100.
+
+
+def asymmetric_l2_loss(u, tau):
+    return torch.mean(torch.abs(tau - (u < 0).float()) * u**2)
+
+
+class IQL(parl.Algorithm):
+    def __init__(self,
```
Comment (zenghsh3) [inline parl/algorithms/torch/iql.py:30]: Pls add `check_model_method` and assert types of arguments as follows:
https://github.com/PaddlePaddle/PARL/blob/20c4bb941e54fd4e221db828d0c1a2c1e1f1ae8a/parl/algorithms/torch/cql.py#L57

**Review by zenghsh3 — APPROVED** [review]
LGTM

File: parl/algorithms/torch/iql.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        policy_loss.backward()
+        self.policy_optimizer.step()
+
+        # Update target Q network
+        self.sync_target(alpha=self.alpha)
+
+        return q_loss.cpu().detach(), v_loss.cpu().detach(), policy_loss.cpu(
+        ).detach()
+
+    def sync_target(self, alpha=0):
```
Comment (TomorrowIsAnOtherDay) [inline parl/algorithms/torch/iql.py:128]: Please add comment for this function.

File: benchmark/torch/iql/train.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            obs, reward, done, _ = env.step(action)
+            avg_reward += reward
+        eval_returns.append(avg_reward)
+    eval_returns = np.array(eval_returns)
+    avg_reward = eval_returns.mean()
+    return avg_reward, eval_returns
+
+
+def main():
+    logger.info("------------------- CQL ---------------------")
```
Comment (TomorrowIsAnOtherDay) [inline benchmark/torch/iql/train.py:49]: CQL -> IQL

**Review by TomorrowIsAnOtherDay — APPROVED** [review]

File: benchmark/torch/iql/train.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        batch_obs, batch_action, batch_reward, batch_next_obs, batch_terminal = rpm.sample_batch(
+            BATCH_SIZE)
+        critic_loss, value_loss, actor_loss = agent.learn(
+            batch_obs, batch_action, batch_reward, batch_next_obs,
+            batch_terminal)
+        tensorboard.add_scalar('train/critic_loss', critic_loss, total_steps)
+        tensorboard.add_scalar('train/value_loss', value_loss, total_steps)
+        tensorboard.add_scalar('train/actor_loss', actor_loss, total_steps)
+        # Evaluate episode
+        if total_steps // args.test_every_steps >= test_flag:
```
Comment (zenghsh3) [inline benchmark/torch/iql/train.py:84]: refine the logic.

**Pull request merged**